app/components-react/shared/ModalLayout.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { ReactNode } from 'react';
+import { useVuex } from '../hooks';
+import { Services } from '../service-provider';
+import { getOS, OS } from '../../util/operating-systems';
+import classNames from 'classnames';
+import { $t } from '../../services/i18n';
+import css from './ModalLayout.m.less';
+
+export interface IModalLayoutProps {
+  hasTitleBar?: boolean;
+  showControls?: boolean;
+  showDone?: boolean;
+  disableDone?: boolean;
+  onSubmit?: (...args: unknown[]) => unknown;
+}
+type TProps = IModalLayoutProps & { children: ReactNode };
+
+/**
+ * A modal layout for showing dialogs
+ */
+export function ModalLayout(partialProps: TProps) {
```

Comment:
is there a reason to prefer es5 `function` syntax over `const ModalLayout = (props) => ...`?

Reply:
I could ask the opposite question. Is there a reason to prefer arrow functions instead of regular ones?

Named functions are always a preferred way to define a function if you don't need to re-bind `this`. Mostly because of 2 reasons.
1) You always see a function name in the debugger call stack instead of anononimous ones
2) Compiler sends the definition of the named function to the top of the file. So if you define `foo()` after `bar()` you can still call `foo()` inside `bar()`

Reply:
I could be wrong, but I think these days chrome treats an anonymous function assigned to a `const` the same as a named function (with the variable name being the name of the function).  Either way, I agree with you and don't have strong preference here.